core/field.js

```diff
@@ -325,6 +334,9 @@ Blockly.Field.prototype.callValidator = function(text) {
   } else if (classResult !== undefined) {
     text = classResult;
   }
+  if (!text.match(Blockly.Field.XML_CHARS_REGEX_)) {
```

Comment:
I don't think this is correct, but it possibly hints at an encoding problem elsewhere. The text (or value) of the field should not be constrained by the the allowed characters in the XML serialization.

If such characters are currently causing a problem, it means we are not properly encoding and decode the text. Usually this is handled by the use of the DOM APIs, and I think (hope?) we're pretty consistent about doing all XML processing through the DOM APIs (instead of manually constructing XML strings).

Reply:
I think the crux of the matter is that `goog.dom.createDom` uses the current document to create the element. Since HTML is more lax than XML, I expect that the browser will accept the string whereas if the target document were XML it would not. Would you consider updating `Blockly.Xml.fieldToDom_` to remove all non-XML characters to be a more appropriate alternative?

Reply:
@ewpatton yes, the main problem is that we're not always using the correct document to create the element.  We should be able to pass the correct document in, we just need to keep a global handle on an XML document that we can use.